File: python/SRS.py
'''
    Simple Rating System
    by: Kyle Galloway

    # A rudimentary college football rating system.
'''
import sys
import time
from includes.graph import Graph
from includes.file_io import FileIO

def main(filename):
    G = Graph()
    file = FileIO(filename)
    G.buildGraph(file)
    print()
    G.rateTeams()
    print()
    teams = G.getTeams()
    teamNames = G.getTeamNames()
    file.printRatings(teamNames,teams)
    print()

if (len(sys.argv) == 2):
    main(sys.argv[1])
elif(len(sys.argv) == 1):
    import os
    for f in os.listdir("csv"):
        main("csv/"+f)
else:
    print("Run with "+sys.argv[0]+" to produce all possible rankings.\n")
    print("Run with "+sys.argv[0]+" <csv/filename> to produce rankings for <filename>.\n")

File: cpp/SRS.cpp
#include <iostream>
#include <string>
#include <time>
#include "StanfordCPPLib/basicgraph.h"
#include "includes/file_io"
using namespace std;

int main(int argc, char *argv[])
{
    /* Grab Filename */
    if(argc != 2) {
        printf("Usage: %s <input filename>\n", argv[0]);
        exit(1);
    }

    /* Make a new graph */
    Graph G = new BasicGraph();

    /* Read in a file */
    FileIO file = new FileIO(argv[1]);

    /* Add the contents of the file to the graph. (Change to local) */
    G.buildGraph(file);
    cout << "\n";

    /* Rate the teams in the graph. (Change to local) */
    G.rateTeams();
    cout << "\n";

    /*
        Get the list of vertexes
        Can be done much more easily using the new basicgraph.h
    */
    Dict teams = G.getTeams();

    /* Get the team names (Possibly unneeded due to better Dict in C++ */
    string[] teamNames = G.getTeamNames();

    /* print the ratings to the output */
    file.printRatings(teamNames,teams);
    cout << "\n";

    return 0;
}